### DISCUSSION TITLE
Navigation item hide sub pages but be able to navigate to them

### DISCUSSION THREAD

**Discussion by OnlyImmutable** [react-navigation/react-navigation#11239] [Q&A]
Hi there,

How can I hide sub pages easily?

```javascript
      <Tab.Screen name={"Rewards"} component={RewardsView} />
      <Tab.Screen name={"Bookings"} component={BookingView} />
      <Tab.Screen name={"Information"} component={InformationView} />
```

I have these 3 pages, I want to add one for profile and one for settings too, but I don't want them to show in my bottom navigation and at present they are, creating an empty space and pushing all the items to the left.

An example can be seen here when I just add profile.

![IMG_5634](https://user-images.githubusercontent.com/19803119/221121194-5e587e46-a966-4af4-9505-75b6927642bd.jpg)

Any guidance would be great as everything online seems to be more focused on disabling the entire bottom bar for specific pages rather than hiding items on the bar but still being able to navigate to them.

**Comment by satya164** [comment]
It doesn't make sense to have a screen that's not in the tab bar. Consider nesting a stack navigator and adding the screen there instead.

**Reply by OnlyImmutable** [reply]
Yes it does? look at instagram, the message list is not accessible via the bottom tab bar? It’s accessible by a clickable icon?

**Reply by satya164** [reply]
> look at instagram, the message list is not accessible via the bottom tab bar? It’s accessible by a clickable icon?

It's a perfect example of what I suggested. The tab bar is also not visible in the message list screen - so it's not part of the tab navigator. It's part of a parent stack navigator.

**Reply by OnlyImmutable** [reply]
There is also the likes of the followers list though which does have the navigation list?

**Reply by satya164** [reply]
> There is also the likes of the followers list though which does have the navigation list?

That's another example. When you're on those screens, the profile tab's icon is highlighted - which means those screens are inside the profile tab (e.g. in a nested stack navigator). If they were just tab navigator screens without a button in the tab bar then nothing would be highlighted in the tab bar.

**Reply by OnlyImmutable** [reply]
How would I layout a stack in this case then? Fairly new to the stack concept

**Reply by satya164** [reply]
You can check https://reactnavigation.org/docs/nesting-navigators/

For the messages list, you'll nest tab inside stack something like:

- RootStack navigator
  - MessagesList screen
  - HomeTabs navigator
    - Feed screen
    - Search screen
    - Profile screen
    - etc.

For the followers, likes etc. you'd need to nest a stack inside the tab:

- RootStack navigator
  - MessagesList screen
  - HomeTabs navigator
    - Feed screen
    - Search screen
    - ProfileStack navigator
      - Profile screen
      - Followers screen
      - etc.
    - etc.

**Reply by OnlyImmutable** [reply]
Do you have a programatic version of that?

**Reply by satya164** [reply]
@OnlyImmutable if you read the fundamentals section in the docs then it'll give you an idea of how to nest things. if you need help then I recommend asking in #help-react-native channel of https://www.reactiflux.com/.
